app/models/user.rb

```diff
@@ -72,9 +72,13 @@ class User < ApplicationRecord
   validate :toxic_email_domain, on: :create
 
   def self.authenticate(who, password)
+    # Avoid exceptions when string is invalid in the given encoding, _or_ cannot be converted
+    # to UTF-8.
+    who = who.encode(Encoding::UTF_8)
```

Comment:
:thinking: Is this the only place facing this problem? Isn't this related to almost any input?

Reply:
This is one of the places it shows up most often, because we are not using controller params to get these

Reply:
Clear, wouldn't make sense to keep this together in controller than?